EventFilter/Utilities/plugins/TcsdRawToDigi.cc

```diff
@@ -0,0 +1,119 @@
+// -*- C++ -*-
+//
+// Package:    ProdTutorial/TcdsRawToDigi
+// Class:      TcdsRawToDigi
+// 
+/**\class TcdsRawToDigi TcdsRawToDigi.cc ProdTutorial/TcdsRawToDigi/plugins/TcdsRawToDigi.cc
+
+ Description:  Producer to unpack lumi nibble from TCDS 
+
+*/
+//
+// Original Author:  Chris Palmer
+//         Created:  Thu, 28 May 2015 19:54:56 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+#include <iostream>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+
+#include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "DataFormats/FEDRawData/interface/FEDHeader.h"
+#include "DataFormats/FEDRawData/interface/FEDTrailer.h"
+#include "DataFormats/FEDRawData/interface/FEDNumbering.h"
+
+#include "EventFilter/FEDInterface/interface/GlobalEventNumber.h"
+#include "EventFilter/FEDInterface/interface/FED1024.h"
+
+#include "DataFormats/FEDRawData/interface/FEDRawData.h"
+
+
+using namespace std;
+
+//
+// class declaration
+//
+
+
+class TcdsRawToDigi : public edm::stream::EDProducer<> {
+   public:
+      explicit TcdsRawToDigi(const edm::ParameterSet&);
+      ~TcdsRawToDigi();
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+   private:
+      virtual void produce(edm::Event&, const edm::EventSetup&) override;
+      
+      int nibble;
```

Comment:
please use a local variable inside the ::produce method implementation.
There is no need to keep it here.